microovn/ovn/bootstrap.go

```diff
@@ -77,14 +74,47 @@ func Bootstrap(s *state.State) error {
 		return fmt.Errorf("Failed to get OVN SB connect string: %w", err)
 	}
 
-	_, err = shared.RunCommand(
-		"ovs-vsctl",
+	nbDB, err := GetOvsdbLocalPath(OvsdbTypeNB)
+	if err != nil {
+		return fmt.Errorf("Failed to get path to OVN NB database socket: %w", err)
+	}
+	sbDB, err := GetOvsdbLocalPath(OvsdbTypeSB)
+	if err != nil {
+		return fmt.Errorf("Failed to get path to OVN SB database socket: %w", err)
+	}
+
+	_, err = NBCtl(
```

Comment:
When connecting to the local DB socket, this command will only succeed when talking to the DB leader.

I see no check for whether this is the leader or not, would it be needed here?

Reply:
If your comment is meant in the context of this specific call to `NBCtl` then I don't think any check is needed as this is a bootstrap operation where it's guaranteed that there are no other cluster members.

If, on the other hand, you are talking about the overall `NBCtl` design, the command does not make any assumptions about remote targets. User is responsible for passing all required arguments, including `--db`, correctly. If there's a problem with execution (e.g. due to executing against non-leader), it'll bubble up as a generic command error.